knimin/lib/data_access.py

```diff
@@ -1588,6 +1588,26 @@ def updateAGKit(self, ag_kit_id, supplied_kit_id, kit_password,
     def updateAGBarcode(self, barcode, ag_kit_id, site_sampled,
                         environment_sampled, sample_date, sample_time,
                         participant_name, notes, refunded, withdrawn):
+        # Get survey ID for participant if needed
+        if participant_name:
+            ag_login_id = self.search_kits(ag_kit_id)[0]
+            sql = """SELECT survey_id
+                     FROM ag_login_surveys
+                     WHERE participant_name = %s AND ag_login_id = %s"""
+            survey_id = self._con.execute_fetchone(
+                sql, [participant_name, ag_login_id])[0]
+        else:
+            survey_id = None
+            participant_name = None
```

Comment:
On closer look, my read is that this would result in an exception in the template?